rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -409,7 +409,7 @@ void SequentialWriter::write(std::shared_ptr<const rosbag2_storage::SerializedBa
     is_first_message_ = false;
   }
 
-  if (should_split_bagfile(message_timestamp)) {
+  if (!storage_options_.snapshot_mode && should_split_bagfile(message_timestamp)) {
```

Comment:
It doesn't make sense to check that we `should_split_bagfile(..)` in snapshot mode because it could be a relatively heavy operation (for instance, for SQLite3). And I don't check it now. 
To print out the logging message, we will have to call `should_split_bagfile(..)` unconditionally first.